### PR TITLE
Only schedule MAC commands in class A downlink slots

### PR DESCRIPTION
**Pull request by rvolosatovs**
#### Summary

References #2909 https://github.com/TheThingsNetwork/lorawan-stack/issues/2774

Incidentally, this makes proceeding with #3052 easier.

cc @ama9910

#### Changes

- Only schedule MAC commands in class A downlink slots
- In case something goes wrong with MAC queue and either responses don't fit or an unknown command is present, this is only logged and does not prevent new MAC requests from being scheduled anymore.

#### Testing

integration testing

##### Regressions

This can affect MAC command scheduling for all classes, regressions not expected.

#### Notes for Reviewers

Actual support for flag from #2909 will be added later.
Perhaps we should implement it as part of the next release to allow users to fallback to the legacy behavior? To be honest, it seems that scheduling MAC in classes B/C causes more trouble than benefits for us and our customers.
In the past we have encountered numerous issues with this and many users were confused about this behavior and consequences it brings to downlink scheduling and priorities. Perhaps we could move #2909 to Backlog instead and only implement if customers explicitly ask for it? Again, as I see it now, I don't think anyone today is going to miss this feature and users are more likely to be happy for this change, however it does change behavior. Thoughts @johanstokking ?

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/networkserver/downlink.go [hunk 1]
```diff
@@ -171,35 +171,42 @@ func (ns *NetworkServer) generateDataDownlink(ctx context.Context, dev *ttnpb.En
...
+			logger := logger.WithField("cid", cmd.CID)
+			desc, ok := spec[cmd.CID]
+			switch {
+			case !ok:
+				logger.Error("Unknown MAC command response enqueued, set FPending")
 				maxDownLen = 0
 				fPending = true
-				break
-			}
-			if desc.DownlinkLength > maxDownLen {
```
Comment (rvolosatovs) [inline pkg/networkserver/downlink.go:187]: Note the bug fixed here - `desc.DownlinkLength` only contains the command payload length, but we should also account for the CID

**Comment by johanstokking** [comment]
> Perhaps we should implement it as part of the next release to allow users to fallback to the legacy behavior? To be honest, it seems that scheduling MAC in classes B/C causes more trouble than benefits for us and our customers.

I agree

> In the past we have encountered numerous issues with this and many users were confused about this behavior and consequences it brings to downlink scheduling and priorities. Perhaps we could move #2909 to Backlog instead and only implement if customers explicitly ask for it?

Yes, fine with me

> Again, as I see it now, I don't think anyone today is going to miss this feature and users are more likely to be happy for this change, however it does change behavior.

I agree

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- MAC commands (both requests and reponses) are now only schduled in class A downlink slots in accordance to latest revisions to LoRaWAN specification.
```
Comment (johanstokking) [inline CHANGELOG.md]:
```suggestion
- MAC commands (both requests and responses) are now only scheduled in class A downlink slots in accordance to latest revisions to LoRaWAN specification.
```

**Review by johanstokking — APPROVED** [review]

**Comment by ama9910** [comment]
Thanks. I don't think the new behaviour will cause any problems for Class C devices that we have implemented/tested, and good to know #2909 is still possible if needed.

**Pull request merged**